### PR TITLE
various: glue for 2.2.0

### PR DESCRIPTION
**Pull request by matildepark**
- Add `local` and `ollama` to the auth command
- Removing `customProvider` and `config.path` in favour of just using pre-existing `model` in our providerConfig and reducing prop passing
- At some point Nolita.ts's `makeAgent` thinks we have too many args, so moved them around to make sense
- serve command doesn't clean up properly (something I fixed in a messy commit in #123), migrated that here
- We no longer see if the API key is longer than 0 chars. If it's a string, we're happy, and we do a `??` check now to not make an empty string `undefined`.
- yargs likes to show you the help command when you bail. This makes the terminal look like a mess with local models. It now doesn't do that.
- contextSize and batchSize need to be set for local models to work at all, so we set them
- adding documentation for local models — and `npx nolita auth` informs users of the article, too, so they understand limitations

### CODE REVIEW

**Comment by matildepark** [comment]
It'd also be nice if `node-llama-cpp` could be less verbose.

File: docs/src/use/local.md
Comment (AlephNotation) [inline docs/src/use/local.md:1]: Maybe add something about `--experimental-vm-modules`` for llama-cpp?

File: tests/agent/generators/generateObject.test.ts
Comment (AlephNotation) [inline tests/agent/generators/generateObject.test.ts:1]: I need to clean this test up a bit. but otherwise good!

**Review by AlephNotation — APPROVED** [review]

File: docs/src/use/local.md
Comment (matildepark) [reply docs/src/use/local.md:1]: can you elaborate? I don't seem to need this flag.

**Pull request merged**